cmake_modules/FindArrow.cmake

```diff
@@ -18,52 +18,82 @@
 # - Find ARROW (arrow/api.h, libarrow.a, libarrow.so)
 # This module defines
 #  ARROW_INCLUDE_DIR, directory containing headers
-#  ARROW_LIBS, directory containing arrow libraries
 #  ARROW_STATIC_LIB, path to libarrow.a
 #  ARROW_SHARED_LIB, path to libarrow's shared library
 #  ARROW_FOUND, whether arrow has been found
 
-if( NOT "$ENV{ARROW_HOME}" STREQUAL "")
-  set(ARROW_HOME "$ENV{ARROW_HOME}")
+if ("$ENV{ARROW_HOME}" STREQUAL "")
+  # PARQUET-955. If the user has set $ARROW_HOME in the environment, we respect
+  # this, otherwise try to locate the pkgconfig in the system environment
+  pkg_check_modules(ARROW arrow)
+else()
+  set(ARROW_FOUND FALSE)
 endif()
 
-set(ARROW_SEARCH_HEADER_PATHS
-  ${ARROW_HOME}/include
-)
+if (ARROW_FOUND)
+  set(ARROW_INCLUDE_DIR ${ARROW_INCLUDE_DIRS})
 
-set(ARROW_SEARCH_LIB_PATH
-  ${ARROW_HOME}/lib
-)
+  if (COMMAND pkg_get_variable)
+    pkg_get_variable(ARROW_ABI_VERSION arrow abi_version)
+  else()
+    set(ARROW_ABI_VERSION "")
+  endif()
+  if (ARROW_ABI_VERSION STREQUAL "")
+    set(ARROW_SHARED_LIB_SUFFIX "")
+  else()
+    set(ARROW_SHARED_LIB_SUFFIX ".${ARROW_ABI_VERSION}")
+  endif()
 
-find_path(ARROW_INCLUDE_DIR arrow/array.h PATHS
-  ${ARROW_SEARCH_HEADER_PATHS}
-  # make sure we don't accidentally pick up a different version
-  NO_DEFAULT_PATH
-)
+  set(ARROW_LIB_NAME ${CMAKE_SHARED_LIBRARY_PREFIX}arrow)
+
+  set(ARROW_SHARED_LIB ${ARROW_LIBDIR}/${ARROW_LIB_NAME}${CMAKE_SHARED_LIBRARY_SUFFIX}${ARROW_SHARED_LIB_SUFFIX})
+  set(ARROW_STATIC_LIB ${ARROW_LIBDIR}/${ARROW_LIB_NAME}${CMAKE_STATIC_LIBRARY_SUFFIX})
+else()
+  if( NOT "$ENV{ARROW_HOME}" STREQUAL "")
```

Comment:
Do we need this check again here? if `ARROW_HOME` is set in the env, `ARROW_FOUND` is always false right?